### ISSUE TITLE
Resources two levels deep inside containers with Freeze URI on don't display properly

### ISSUE THREAD

**Issue by jcamachott** [modxcms/revolution#13020]
### Summary

Resources two levels deep inside containers with Freeze URI on don't display properly
### Environment

MODX 2.5
friendly_urls Yes
friendly_urls_strict No
use_alias_path Yes
use_frozen_parent_uris Yes

I have MODX installed in the root of my site. I have a Home resource, freeze URI is set to "/". Inside Home I have some resources one of which is also a container, so it looks like this:

Home (container)
-> Disclaimer
-> Privacy
-> Users (container)
----> Login
----> Register

The Disclaimer page views properly at domain.com/disclaimer.

There are two problems:
1. Smaller problem: When I view the Login page, in the address bar I see domain.com//users/login (two slashes) but more importantly...
2. Bigger problem: The content that I see is the 'front' template which is attached ONLY to the Home page. Why would this happen?

Now if I move the Login document outside the Users container (and after clearing cache), it displays fine.

So what I needed to do was remove the Freeze URI from the Home resource and change the Freeze URI to ON for Users (set to 'users/') and for all the child resources (because I want to see /disclaimer rather than /home/disclaimer)...and now domain.com/users/login works.

**Comment by Mark-H** [comment]
Maybe it's showing your front template/home page because the double slash URL can't be found, which gets send to the error page, while you don't have a proper error page set up?

**Comment by jcamachott** [comment]
Yes its the error page....but why would the double slashes show up anyway in the address (right clicking on Login page and choosing View in MODX).

**Comment by JaneEve** [comment]
I have same problem.. have you find a solution?

**Comment by Ibochkarev** [comment]
### Triage (current 3.x)

Still looks valid. No fix PR. Labels: `bug` only.

### Comments

@Mark-H guessed the Home/"front" template was the error page because `//users/login` never matched a resource. @jcamachott confirmed that. @JaneEve (2018) still saw the same double-slash case.

### Current code

With `use_frozen_parent_uris` and a parent frozen to `/`, `getAliasPath()` does `rtrim($uri, '/')`, pushes the empty string, then `implode('/', …)`. That yields a leading `/` on the child URI (e.g. `/users/login`). Prefixed with the site URL that becomes `//users/login`, which fails alias lookup and falls through to the error page.

Nothing in that branch skips an empty segment after stripping `/`. I did not run a live manager tree for this setup; the path above matches the report.

### Suggested work

When taking a frozen parent URI, skip adding the segment if `rtrim($uri, '/')` is empty (treat `/` as site root). Optionally normalize so generated `uri` values never start with `/` when the alias map expects unprefixed paths.

Worth keeping open until that lands (or someone confirms a live 3.x site no longer reproduces).
